### PR TITLE
Cleanup rollingstock model

### LINKED ISSUE #6578: Cleanup rolling stock supported signaling systems
## Description

The signaling system list should be considered as dynamic information loaded from "extension".

Editoast validates the `supported_signaling_systems` using an `enum` which is not valid.

## What should be done?

1. Remove the enum: https://github.com/osrd-project/osrd/blob/5895b8726a4f1969fdcd922404ac7aa2e388dcb2/editoast/src/schema/rolling_stock/mod.rs#L268-L277
2. Replace where the enum was used by `Vec<String>`
3. Remove the `feature` unused field:
  - Make a migration
  - Adapt editoast
  - Adapt core
  - Adapt python `osrd_schemas`

### PR DESCRIPTION
**Pull request by SarahBellaha**
closes #6578

### CODE REVIEW

**Review by eckter — APPROVED** [review]
LGTM on core / python

(there's a flake8 error to be fixed first though)

**Review by clarani — APPROVED** [review]
LGTM for the frontend part

:warning: you have a linter error which breaks the build

**Comment by clarani** [comment]
Maybe you should merge your 2 commits into 1

File: editoast/migrations/2024-02-06-171107_remove_features_from_rollingstock/up.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+ALTER TABLE rolling_stock
+DROP COLUMN features,
+ALTER COLUMN supported_signaling_systems DROP DEFAULT,
+ALTER COLUMN supported_signaling_systems TYPE TEXT[] USING TRANSLATE(supported_signaling_systems::jsonb::text, '[]','{}')::TEXT[],
+ADD CONSTRAINT supported_signaling_systems_no_nulls CHECK (array_position(supported_signaling_systems, NULL) IS NULL),
+ALTER COLUMN supported_signaling_systems SET DEFAULT ('{"BAPR", "BAL", "TVM300", "TVM430"}');
```
Comment (flomonster) [inline editoast/migrations/2024-02-06-171107_remove_features_from_rollingstock/up.sql]: We don't need the constraint and a default value.

```suggestion
ALTER TABLE rolling_stock DROP COLUMN features,
    ALTER COLUMN supported_signaling_systems DROP DEFAULT,
    ALTER COLUMN supported_signaling_systems TYPE TEXT [] USING TRANSLATE(
        supported_signaling_systems::jsonb::text, '[]', '{}')::TEXT [];
```

File: editoast/migrations/2024-02-06-171107_remove_features_from_rollingstock/down.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ALTER TABLE rolling_stock
+ADD features text[] NOT NULL DEFAULT '{}',
+ALTER COLUMN supported_signaling_systems DROP DEFAULT,
+ALTER COLUMN supported_signaling_systems type jsonb USING Array_to_json(supported_signaling_systems::text[]),
+DROP CONSTRAINT supported_signaling_systems_no_nulls,
+ALTER COLUMN supported_signaling_systems SET DEFAULT ('["BAPR", "BAL", "TVM300", "TVM430"]');
```
Comment (flomonster) [inline editoast/migrations/2024-02-06-171107_remove_features_from_rollingstock/down.sql]:
```suggestion
ALTER TABLE rolling_stock
ADD features text [] NOT NULL DEFAULT '{}',
    ALTER COLUMN supported_signaling_systems type jsonb USING Array_to_json(supported_signaling_systems::text []),
    ALTER COLUMN supported_signaling_systems
SET DEFAULT ('["BAPR", "BAL", "TVM300", "TVM430"]');
```

**Review by flomonster — APPROVED** [review]
LGTM (review and tested). Only one suggestion to simplify migrations.

**Review by leovalais — APPROVED** [review]

**Pull request merged**